src/library_async.js

```diff
@@ -181,11 +187,19 @@ mergeInto(LibraryManager.library, {
       return func;
     },
 
+    doRewind: function(ptr) {
+      var start = Asyncify.getDataRewindFunc(ptr);
+#if ASYNCIFY_DEBUG
+      err('ASYNCIFY: start:', start);
+#endif
+      // Once we have rewound the stack the runtime could potentially exist
```

Comment:
But I'm not sure this comment is true. How could it exit at this time? Nothing else happened while we were sleeping, so nothing happened that would allow us to exit.

But the code is right - we need to keep ourselves alive during the sleep.

Reply:
I guess I mean we want to allow the runtime to exit once the user code is running again..   maybe this sentence isn't clear?

Reply:
Yes, I think that's what's confusing me. We want to allow it to exit eventually, but it won't exit right here.

Reply:
The new comment is maybe more clear?

```
      // Once we have rewound and the stack we no longer need to artificially keep
      // the runtime alive.
```

Reply:
Thanks, yes.